monai/transforms/transforms.py

```diff
@@ -441,10 +444,514 @@ def __call__(self, img):
+            img = torch.tensor(img)
+        if not torch.is_tensor(grid):
+            grid = torch.tensor(grid)
+        if self.device:
+            img = img.to(self.device)
+            grid = grid.to(self.device)
+
+        for i, dim in enumerate(img.shape[1:]):
+            grid[i] = 2. * grid[i] / (dim - 1.)
+        grid = grid[:-1] / grid[-1:]
+        grid = grid[range(img.ndim - 2, -1, -1)]
+        grid = grid.permute(list(range(grid.ndim))[1:] + [0])
+        out = torch.nn.functional.grid_sample(img[None].float(),
+                                              grid[None].float(),
+                                              mode=mode,
+                                              padding_mode=self.padding_mode,
+                                              align_corners=False)[0]
+        if not self.as_tensor_output:
+            return out.cpu().numpy()
+        return out
+
+
+@export
+class Affine:
+    """
+    transform ``img`` given the affine parameters.
+    """
+
+    def __init__(self,
+                 rotate_params=None,
+                 shear_params=None,
+                 translate_params=None,
+                 scale_params=None,
+                 padding_mode='zeros',
+                 as_tensor_output=False,
+                 device=None):
+        """
+        The affines are applied in rotate, shear, translate, scale order.
+
+        Args:
+            rotate_params (float, list of floats): a rotation angle in radians,
+                a scalar for 2D image, a tuple of 2 floats for 3D. Defaults to no rotation.
+            shear_params (list of floats):
+                a tuple of 2 floats for 2D, a tuple of 6 floats for 3D. Defaults to no shearing.
+            translate_params (list of floats):
+                a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Translation is in pixel/voxel
+                relative to the center of the input image. Defaults to no translation.
+            scale_params (list of floats):
+                a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Defaults to no scaling.
+            padding_mode ('zeros'|'border'|'reflection'): mode of handling out of range indices. Defaults to 'zeros'.
+            as_tensor_output (bool): the computation is implemented using pytorch tensors, this option specifies
+                whether to convert it back to numpy arrays.
+            device (torch.device): device on which the tensor will be allocated.
+        """
+        self.affine_grid = AffineGrid(rotate_params,
+                                      shear_params,
+                                      translate_params,
+                                      scale_params,
+                                      as_tensor_output=True,
+                                      device=device)
+        self.resampler = Resample(padding_mode, as_tensor_output=as_tensor_output, device=device)
+
+    def __call__(self, img, spatial_size, mode='bilinear'):
```

Comment:
Can we use the same **spatial_size** and **mode** for all images in training?
If yes, strongly suggest to move them to __init__(), then we can compose this transform with others as a transform chain.
Otherwise, I feel it's not easy to use it with other transforms together.
Thanks.